cpp/src/io/comp/comp.cpp

```diff
@@ -339,29 +339,34 @@ void host_compress(compression_type compression,
     return h_inputs[a].size() > h_inputs[b].size();
   });
 
-  std::vector<std::future<size_t>> tasks;
+  auto h_results =
+    cudf::detail::make_pinned_vector_async<compression_result>(results.size(), stream);
+  cudf::detail::cuda_memcpy<compression_result>(h_results, results, stream);
+
+  std::vector<std::future<std::pair<size_t, size_t>>> tasks;
   auto const num_streams =
     std::min<std::size_t>(num_chunks, cudf::detail::host_worker_pool().get_thread_count());
   auto const streams = cudf::detail::fork_streams(stream, num_streams);
   for (size_t i = 0; i < num_chunks; ++i) {
     auto const idx        = task_order[i];
     auto const cur_stream = streams[i % streams.size()];
-    auto task =
-      [d_in = h_inputs[idx], d_out = h_outputs[idx], cur_stream, compression]() -> size_t {
+    if (h_results[task_order[i]].status == compression_status::SKIPPED) { continue; }
+
+    auto task = [d_in = h_inputs[idx], d_out = h_outputs[idx], cur_stream, compression, idx]() {
```

Comment:
Is there value in specifying the lambda's return type?  Or is that only for `__device__` lambdas? 
(I see that we've removed the ` -> size_t`.)

Reply:
There is only one return in the lambda and it already includes the type. 
I see no value is specifying the type twice.

Reply:
yeah, I'm not sure why the return type was there. I don't specify it when I don't have to 🤷